### DISCUSSION TITLE
Have I set it up correctly?

### DISCUSSION THREAD

**Discussion by daath** [johanmeijer/grott#329] [General]
If you haven't seen, I started by setting Grott up completely wrong.

After having found the guide to setting it up and integrating it to HA, I have questions.

I have to inverters. Both talk to Grott and it relays the messages to the growatt servers :) So far so good.

Here is what I have done so far:

I created one instance of Grott in Portainer, running on default port 5279 - it works.
The only environment variables in the container I have set are PUID, PGID and TZ - the rest is set in grott.ini
I have two volumes set: /docker/grott/grott.ini:/app/grott.ini and /docker/grott/grott_ha.py:/app/grott_ha.py 
I added the grott_ha extension in grott.ini - from what I can see in the logs, it runs it, so it reads the grott.ini file

Modified my data loggers to talk to Grott - it works

I followed example 2 here: https://github.com/johanmeijer/grott/wiki/Home-Assistant-configuration-examples
as well as the docs here: https://github.com/johanmeijer/grott/tree/master/examples/Home%20Assistent
However, the example is wrong. There is no sensors_grott_eng.yaml - there is an mqtt_grott_eng.yaml and that has to be added differently - it needs to be under "mqtt:" not just "sensor:" ...

When I look in MQTT I can see that it only uses a single topic: energy/growatt - I read [here](https://github.com/johanmeijer/grott/wiki/Multi-Inverter-support) that it should use something like energy/growatt/DEVICEID ...

In the Grott logs, I also see this:

```
2023-04-20T17:51:18.301916199Z 	 - Grott extension processing started :  grott_ha
2023-04-20T17:51:18.303227606Z 	[Grott HA] 0.0.7-rc6 Can't configure device: XXXXXXXXXX 
2023-04-20T17:51:18.303799172Z 	 - Grott extension processing ended :  7
```
Where XXXXXXXXXX is the serial for my MIC 3000TL-X inverter.

I did follow the docs and set invtypemap in grott.ini - docs here: https://github.com/johanmeijer/grott/wiki/Multi-Inverter-support

In HA I see 18 entities - nowhere near as many as I expected, and Energy Generated (today/total) is "Unknown" ... 

I hope you got through this mess, and that you have some insights for me :)

Oh, and I also put this in my configuration.yaml:

```
homeassistant:
  customize_glob:
    sensor.growatt_*_energy*:
      last_reset: '1970-01-01T00:00:00+00:00'
      device_class: energy
      state_class: measurement

```

Thanks in advance!

**Comment by daath** [comment]
Here is a MQTT message that came from the MIC 3000TL-X inverter (ShineWifi-X):

`{
  "device": "XXXXXXXXXXX",
  "time": "2023-04-20T19:36:01",
  "buffered": "no",
  "values": {
    "#recortype": 3124,
    "pvstatus": 1,
    "pvpowerin": 2166,
    "pv1voltage": 2907,
    "pv1current": 7,
    "pv1watt": 2166,
    "pv2voltage": 0,
    "pv2current": 0,
    "pv2watt": 0,
    "pvpowerout": 0,
    "pvfrequentie": 0,
    "pvgridvoltage": 2360,
    "pvgridcurrent": 0,
    "pvgridpower": 0,
    "pvgridvoltage2": 0,
    "pvgridcurrent2": 0,
    "pvgridpower2": 0,
    "pvgridvoltage3": 0,
    "pvgridcurrent3": 13,
    "pvgridpower3": 3116302336,
    "totworktime": 1097,
    "eactoday": 1097,
    "eactotal": 155,
    "epvtotal": 4587520,
    "epv1today": 0,
    "epv1total": 0,
    "epv2today": 0,
    "epv2total": 0,
    "pvtemperature": 471,
    "pvipmtemperature": 0,
    "pvboosttemp": 0,
    "bat_dsp": 503,
    "pbusvolt": 3620,
    "#nbusvolt": 0,
    "#ipf": 20000,
    "#realoppercent": 7,
    "#opfullwatt": 0,
    "#deratingmode": 0,
    "eacharge_today": 0,
    "eacharge_total": 0,
    "batterytype": 0,
    "uwsysworkmode": 0,
    "systemfaultword1": 0,
    "systemfaultword2": 0,
    "systemfaultword3": 0,
    "systemfaultword4": 0,
    "systemfaultword5": 0,
    "systemfaultword6": 0,
    "systemfaultword7": 0,
    "pdischarge1": 0,
    "p1charge1": 0,
    "vbat": 0,
    "SOC": 0,
    "pactouserr": 0,
    "#pactousers": 0,
    "#pactousert": 0,
    "pactousertot": 0,
    "pactogridr": 0,
    "#pactogrids": 0,
    "#pactogridt": 0,
    "pactogridtot": 0,
    "plocaloadr": 0,
    "#plocaloads": 0,
    "#plocaloadt": 0,
    "plocaloadtot": 0,
    "#ipm": 0,
    "#battemp": 0,
    "spdspstatus": 0,
    "spbusvolt": 0,
    "etouser_tod": 0,
    "etouser_tot": 0,
    "etogrid_tod": 0,
    "etogrid_tot": 0,
    "edischarge1_tod": 0,
    "edischarge1_tot": 0,
    "eharge1_tod": 0,
    "eharge1_tot": 0,
    "elocalload_tod": 0,
    "elocalload_tot": 0
  }
}
`

**Comment by daath** [comment]
I mean ... Some of it works ;P This is the SPH3600 inverter:
![image](https://user-images.githubusercontent.com/543124/233483609-aae943f3-1941-411e-83e4-c3a19db2a9ad.png)

But no energy produced today/total and no second inverter - the MIC 3000TL-X (which is of course offline now that the sun has set - but it wasn't there in full sun either) ... :/

**Comment by daath** [comment]
OK, I am closer now - I had this in my docker: ledidobe/grottrpi - I changed it to ledidobe/grottrpi:beta and now I have the topic I expected :P

Let's see tomorrow if my other inverter comes on - and if Energy Dashboard works, and all that other jazz :)

**Comment by daath** [comment]
OK, everything more or less works. Had I only known to add ":beta" to my docker image, many hours could have been saved 🤣

**Comment by johanmeijer** [comment]
Sorry for that. I will promote the beta to the latest soon.......

**Reply by daath** [reply]
No worries :) Works well! Only thing I can't figure out is how to get the battery SoC ... :)
